File: test_python/soal4_test.py
import unittest
import sys
import os

current_dir = os.path.dirname(os.path.abspath(__file__))

target_path = os.path.join(current_dir, '..', 'python', 'soal4')

sys.path.append(os.path.abspath(target_path))

from soal4 import maskify

class TestMaxTriSum(unittest.TestCase):
    def test_examples(self):
        self.assertEqual(maskify("4556364607935616"), "############5616")
    def test_examples1(self):
        self.assertEqual(maskify("64607935616"), "#######5616")
    def test_examples2(self):
        self.assertEqual(maskify("1"), "1")

    def test_edge_cases(self):
        self.assertEqual(maskify(""), "")
    def test_edge_cases1(self):
        self.assertEqual(maskify("Skippy"), "##ippy")
    def test_edge_cases2(self):
        self.assertEqual(maskify("Nananananananananananananananana Batman!"), 
                         "####################################man!")
    def test_edge_cases6(self):
        self.assertEqual(maskify("1234567890"), "######7890")
    def test_edge_cases3(self):
        self.assertEqual(maskify("abcd"), "abcd")
    def test_edge_cases4(self):
        self.assertEqual(maskify("a"), "a")
    def test_edge_cases5(self):
        self.assertEqual(maskify("123"), "123")

if __name__ == '__main__':
    unittest.main()

File: test_php/soal4_test.php
<?php

use PHPUnit\Framework\TestCase;

require_once dirname(__DIR__) . '/php/soal4/soal4.php';
class Soal4_Test extends TestCase
{
    public function testExamples() {
        $this->assertSame("############5616", maskify("4556364607935616"));
    }
    public function testExamples2() {
        $this->assertSame("#######5616", maskify("64607935616"));
    }
    public function testExamples3() {
        $this->assertSame("1", maskify("1"));
    }
    public function testExamples4() {
        $this->assertSame("", maskify(""));
    }
    public function testExamples5() {
        $this->assertSame("##ippy", maskify("Skippy"));
    }
    public function testExamples6() {
        $this->assertSame("####################################man!", maskify("Nananananananananananananananana Batman!"));
    }

    public function testAdditional() {
        $this->assertSame("######7890", maskify("1234567890"));
    }
    public function testAdditional2() {
        $this->assertSame("abcd", maskify("abcd"));
    }
    public function testAdditional3() {
        $this->assertSame("a", maskify("a"));
    }
    public function testAdditional4() {
        $this->assertSame("123", maskify("123"));
    }
}
